File: cmps11/lab7/GCD.c
/*Ian Feekes
 *#1474914
 *CMPS 11
 *GCD.c
 *Computes the greatest common divisor of two inputs by the user through C
*/
#include<stdio.h>
#include<stdlib.h>
#include <math.h>

int main()
{
 int int1=0;
 int int2=0;
 printf("Enter a positive integer: ");
 scanf("%i",&int1);
 while(int1<=0) 
 {
  printf("Please enter a positive integer: ");
  scanf("%i",&int1);
 }
 printf("Enter another positive integer: ");
 scanf("%i",&int2);
 while (int2<=0)
 {
  printf("Please enter a positive integer: ");
  scanf("%i",&int2);
 } 
 int divisor=0;
 int divisorA=0;
 int remainder=0;
 if(int1>=int2)
 {
  divisor=int2;
  remainder=int1%int2;
 }
 else
 {
  divisor=int1;
  remainder=int2%int1;
 }
 while(remainder!=0)
 {
  divisorA=remainder;                     
  remainder=divisor%remainder;
  divisor=divisorA;
 } 
 printf("The GCD of %i and %i is %i \n",int1, int2, divisor);
 return 0;
} 

File: cmps11/PA3/GCD.java
//PA3 GCD
//Ian Feekes
//1474914
//CMPS 11
import java.util.Scanner;
public class GCD
{
 public static void main(String[] args)
 {
  Scanner scan = new Scanner(System.in);           //I like the name scan more for organizational purposes
  System.out.println("Enter a positive integer: ");
  boolean loopKeeper=true;                        //Used to verify that the user enters a positive integer
  int firstInt=0;				  //Initialized to zero because program bugged out otherwise
  int  secondInt=0;
  while(loopKeeper==true)                        //Double while loop firstly ensures they enter an integer and seondly ensures the integer is positive
  {
   while(!scan.hasNextInt())
   {
    scan.next();
    System.out.println("Please enter a positive integer: ");
   }
   firstInt=scan.nextInt();
   if(firstInt>0) loopKeeper=false;
   else System.out.println("Please enter a positive integer: ");
  }
  System.out.println("Enter another positive integer: ");
  while(loopKeeper==false)
  {
   while(!scan.hasNextInt())
   {
    scan.next();
    System.out.println("Please enter a positive integer: ");
   }
   secondInt=scan.nextInt(); 
   if(secondInt>0) loopKeeper=true;
   else System.out.println("Please enter a positive integer: ");
  }
  int divisorA=0;                                //divisorA is used to keep place for dividing purposes
  int divisor=0;				 //divisor and remainder are used when finding the lowest common denominator
  int remainder=0;
  if(firstInt>=secondInt)
  {
   divisor=secondInt;
   remainder=firstInt%secondInt;
  }
  else 
  {
   remainder=secondInt%firstInt;
   divisor=firstInt;
  }
  while(remainder!=0)
  {
   divisorA=remainder;                          //because of the while loop it will keep doing this until a GCD is found, even if it is 1
   remainder=divisor%remainder;
   divisor=divisorA;
  }
  System.out.println("The GCD of "+firstInt+" and "+secondInt+" is "+divisor);
 }
} 
